solidity/contracts/token/HypERC20Collateral.sol

```diff
@@ -75,6 +83,8 @@ contract HypERC20Collateral is TokenRouter {
         uint256 _amount,
         bytes calldata // no metadata
     ) internal virtual override {
+        sdeUSD.cooldownAssets(_amount);
+        sdeUSD.unstake(address(this));
         wrappedToken.safeTransfer(_recipient, _amount);
```

Comment:
i think we can remove the `safeTransfer` and just have the user address be the parameter of the `unstake` call

Reply:
im just worried that there may be a chance that the underlyingAmount when unstaking is > _amount, but do you think that invariant holds?

Reply:
I think that should hold, if it doesnt it means that sdeusd was hacked (collteral staked was drained so value of shares are less than 1)